### ISSUE TITLE
Custom remark & rehype plugins for markdown?

### ISSUE THREAD

**Issue by husayt** [vitejs/vite-plugin-react-pages#159]
Is there a way to add custom remark & rehype plugins for markdown? seems they are hardcoded at the moment.

for example, something similar to what vitepress provides [here](https://vitepress.dev/guide/markdown#advanced-configuration)

**Comment by csr632** [comment]
@husayt Implemented in https://github.com/vitejs/vite-plugin-react-pages/pull/160/files and published as `vite-plugin-react-pages@4.1.5-alpha.0`. You can try it out :)

**Comment by husayt** [comment]
This is really exciting. Can you please, provide an example of config?

**Comment by husayt** [comment]
this is what I ended up using in viteconfig

```

export default defineConfig({
  plugins: [
    react(),
    pages({
      pagesDir: path.join(__dirname, 'pages'),
      /** user can add/remove remark plugins passed to mdx */
      modifyRemarkPlugins: [
        {
          name: 'remark-emoji',
          createPlugin: () => import('remark-emoji').then((m) => m.default),
        },
        {
          name: 'emark-mdc',
          createPlugin: () => import('demark-mdc').then((m) => m.default),
        },
        {
          name: 'emark-mdc',
          createPlugin: () => import('demark-mdc').then((m) => m.default),
        },
        // {
        //   name: 'remark-rehype',
        //   createPlugin: () => import('remark-rehype').then((m) => m.default),
        // },
      ],
      /** user can add/remove rehype plugins passed to mdx */
      modifyRehypePlugins: [
        {
          name: 'rehype-external-links',
          createPlugin: () =>
            import('rehype-external-links').then((m) => m.default),
        },

        {
          name: 'rehype-raw',
          createPlugin: () => import('rehype-raw').then((m) => m.default),
        },
        {
          name: 'rehype-sort-attribute-values',
          createPlugin: () =>
            import('rehype-sort-attribute-values').then((m) => m.default),
        },
        {
          name: 'rehype-sort-attributes',
          createPlugin: () =>
            import('rehype-sort-attributes').then((m) => m.default),
        },
      ],
    }),
  ],
})
```

**Comment by csr632** [comment]
Example: 
```

export default defineConfig({
  plugins: [
    react(),
    pages({
      pagesDir: path.join(__dirname, 'pages'),
      modifyRemarkPlugins(original) {
        return [
          ...original,
          {
            name: 'remark-emoji',
            createPlugin: () => import('remark-emoji').then((m) => m.default),
          }
        ]
      },
      modifyRehypePlugins(original) {
        return [
          ...original,
          // add rehype plugins here
        ]
      },
    }),
  ],
})

```

**Comment by csr632** [comment]
Landed in `vite-plugin-react-pages@4.1.5`
